src/ensembl/src/content/app/species-selector/state/speciesSelectorActions.ts

```diff
@@ -291,6 +296,12 @@ export const deleteSpeciesAndSave: ActionCreator<
 
   if (deletedSpecies) {
     const deletedSpeciesName = getSpeciesAnalyticsName(deletedSpecies);
+
+    analyticsTracking.setCustomDimension({
+      diemension: CustomDimensions.SPECIES,
+      value: deletedSpecies.genome_id
+    });
```

Comment:
to middleware?

Reply:
Created separate setter functions for dimensions as discussed.